tools/debug-ui/src/BlocksView.tsx

```diff
@@ -0,0 +1,188 @@
+import './BlocksView.scss';
+import { useQuery } from 'react-query';
+import { fetchChainProcessingStatus, BlockProcessingStatus, ChunkProcessingStatus } from './api';
+
+type BlocksViewProps = {
+    addr: string;
+};
+
+function prettyTime(datetime: string) {
+    const time = new Date(Date.parse(datetime));
+    return (
+        time.getUTCHours() +
+        ':' +
+        String(time.getUTCMinutes()).padStart(2, '0') +
+        ':' +
+        String(time.getUTCSeconds()).padStart(2, '0') +
+        '.' +
+        String(time.getUTCMilliseconds()).padStart(3, '0')
+    );
+}
+
+function printTimeInMs(time: number | null) {
+    if (time == null) {
+        return 'N/A';
+    } else {
+        return time + 'ms';
+    }
+}
+
+function printDuration(start: string, end: string): string {
+    const duration = Date.parse(end) - Date.parse(start);
+    if (duration > 0) {
+        return '+' + duration + 'ms';
+    } else {
+        return duration + 'ms';
+    }
+}
+
+function getChunkStatusSymbol(chunkStatus: ChunkProcessingStatus): string {
+    switch (chunkStatus) {
+        case 'Completed':
+            return '✔';
+        case 'Requested':
+            return '⬇';
+        case 'NeedToRequest':
+            return '.';
+        default:
+            return '';
+    }
+}
+
+function printBlockStatus(blockStatus: BlockProcessingStatus): string {
+    switch (blockStatus) {
+        case 'Orphan':
+            return 'Orphan';
+        case 'WaitingForChunks':
+            return 'WaitingForChunks';
+        case 'InProcessing':
+            return 'InProcessing';
+        case 'Accepted':
+            return 'Accepted';
+        case 'Unknown':
+            return 'Unknown';
+        case 'HeightProcessed':
+            return 'HeightProcessed';
+        case 'TooManyProcessingBlocks':
+            return 'TooManyProcessingBlocks';
+        case 'Error':
+            return 'Error';
+        default:
+            return '';
+    }
+}
+
+export const BlocksView = ({ addr }: BlocksViewProps) => {
+    const {
+        data: chainProcessingInfo,
+        error: chainProcessingInfoError,
+        isLoading: chainProcessingInfoLoading,
+    } = useQuery(['chainProcessingInfo', addr], () => fetchChainProcessingStatus(addr));
+    if (chainProcessingInfoLoading) {
+        return <div>Loading...</div>;
+    } else if (chainProcessingInfoError) {
+        return (
+            <div className="chain-and-chunk-blocks-view">
+                <div className="error">{(chainProcessingInfoError as Error).stack}</div>
+            </div>
+        );
+    } else if (!chainProcessingInfo) {
+        return (
+            <div className="chain-and-chunk-blocks-view">
+                <div className="error">No Data</div>
+            </div>
+        );
+    }
+    let numShards = 0;
```

Comment:
nit: can just initialize numShards with the value you're adding after. Also this seems like it can be const.

Reply:
Cool, wasn't sure if the next statement was guaranteed since it may result in an error and Typescript might not accept that, but looks like you're right. 
Updated as suggested.